build/archives/BUILD.gn

```diff
@@ -48,17 +48,36 @@ if (host_os == "linux") {
   }
 }
 
+# Generates flutter_patched_sdk.zip for debug and profile runtime modes
+# and saves it in the zips_archive folder. For release runtime mode it
+# generates flutter_patched_sdk_product.zip and saves it to
+# zips_archive/flutter_patched_sdk_product/ folder.
 zip_bundle("flutter_patched_sdk") {
-  output = "flutter_patched_sdk.zip"
+  file_suffix = ""
+  path_prefix = ""
+  if (flutter_runtime_mode == "release") {
+    file_suffix = "_product"
+    path_prefix = "flutter_patched_sdk_product/"
+  }
+  output = "flutter_patched_sdk${file_suffix}.zip"
   deps = [ "//flutter/lib/snapshot:strong_platform" ]
   files = [
     {
       source = "$root_out_dir/flutter_patched_sdk/vm_outline_strong.dill"
-      destination = "vm_outline_strong.dill"
+      destination = "${path_prefix}vm_outline_strong.dill"
     },
     {
       source = "$root_out_dir/flutter_patched_sdk/platform_strong.dill"
-      destination = "platform_strong.dill"
+      destination = "${path_prefix}platform_strong.dill"
     },
   ]
 }
+
+# Flutter consumes the dart sdk as a prebuilt. Rather than regenerating
+# the zip file we are just copying the original file to the artifacts location.
+if (build_engine_artifacts) {
+  copy("dart_sdk_archive") {
+    sources = [ prebuilt_dart_sdk ]
+    outputs = [ "$root_out_dir/zip_archives/dart-sdk-$host_os-$target_cpu.zip" ]
```

Comment:
`prebuilt_dart_sdk` is a directory, so I'm not quite sure why GN isn't complaining about this. Maybe it's because the build isn't trying to build the target yet. Also `$host_os-$target_cpu` should probably be host-host or target-target. Here's a suggestion on how to make sure things stay consistent:

```gn
if (build_engine_artifacts && flutter_prebuilt_dart_sdk) {
  copy("dart_sdk_archive") {
    sources = [ prebuilt_dart_sdk_archive ]
    outputs = [ "$root_out_dir/zip_archives/dart-sdk-$prebuilt_dart_sdk_config.zip" ]
  }
}
```

`prebuilt_dart_sdk_archive` and `prebuilt_dart_sdk_config` we can define in `common/config.gni`.

Reply:
done!